### PR TITLE
Paged archives

### PR DESCRIPTION
**Pull request by dakota**
This adds support for paging archiving worker

### CODE REVIEW

File: src/Worker/ArchiveWorker.php [hunk 1]
```diff
@@ -44,6 +48,95 @@ protected function _ensureTable(Table $archiveTable)
...
+    {
+        $baseQuery = $this->DelayedJobs->query()
+            ->where(['status IN' => [Job::STATUS_BURIED, Job::STATUS_SUCCESS]])
+            ->order(['id' => 'ASC']);
+
+        $total = $baseQuery->count();
+        if ($total === 0) {
+            Log::debug('No jobs to be archived.');
+
+            return;
```
Comment (johanmeiring) [inline src/Worker/ArchiveWorker.php:102]: This seems incompatible with the return type indicated in the method signature, and will likely throw a fatal error in the almost impossible event that there are no jobs to archive.

Comment (johanmeiring) [reply src/Worker/ArchiveWorker.php:102]: Either that, or my inexperience with `\Generator` is showing :-)

File: src/Worker/ArchiveWorker.php [hunk 2]
```diff
@@ -44,6 +48,95 @@ protected function _ensureTable(Table $archiveTable)
...
+                ->select(['id'])
+                ->where(['created <=' => $time])
+                ->orderDesc('id')
+                ->disableHydration()
+                ->first()['id'] ?? null;
+
+        if (!$lastIdToDelete) {
+            Log::debug('No jobs to be deleted.');
+
+            return;
```
Comment (johanmeiring) [inline src/Worker/ArchiveWorker.php:126]: Same as before, but odds are good I don't fully understand how this works.

File: src/Worker/ArchiveWorker.php [hunk 3]
```diff
@@ -54,40 +147,55 @@ public function __invoke(Job $job)
             return 'Not enabled for archiving';
         }
 
-        $delayedJobsTable = TableRegistry::getTableLocator()->get('DelayedJobs.DelayedJobs');
-        $archiveTable = TableRegistry::getTableLocator()->get('Archive', [
+        $this->loadModel('DelayedJobs.DelayedJobs');
+        $this->Archive = TableRegistry::getTableLocator()->get('Archive', [
```
Comment (johanmeiring) [inline src/Worker/ArchiveWorker.php:151]: Isn't TableRegistry deprecated now, or in 4.1?  Either way, might make sense to not use it?

File: src/Worker/ArchiveWorker.php (hunk 1 above)
Comment (dakota) [reply src/Worker/ArchiveWorker.php:102]: This is actually correct! Took me a while to understand the documentation, but basically if you use `yield` the return type is Generator (always). Anything you do `return` is available via the `getReturn()` method on the generator object (that is actually returned by calling the generator method)

File: src/Worker/ArchiveWorker.php (hunk 3 above)
Comment (dakota) [reply src/Worker/ArchiveWorker.php:151]: In 4.1, you are right, will replace it

File: src/Worker/ArchiveWorker.php (hunk 1 above)
Comment (johanmeiring) [reply src/Worker/ArchiveWorker.php:102]: Ah great, understood :+1:

File: src/Worker/ArchiveWorker.php (hunk 3 above)
Comment (dakota) [reply src/Worker/ArchiveWorker.php]: @johanmeiring Changes made

**Pull request merged**